test/json.test.ts

```diff
@@ -0,0 +1,270 @@
+    });
+
+    test("single-line comment after string with escaped double quote", () => {
+        const text =
+            `{
+    "hello": "world\\"" // comment
+}`;
+
+        const expected =
+            `{
+    "hello": "world\\""
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("multi-line comment at start of text", () => {
+        const text =
+            `/**/{
+    "hello": "world"
+}`;
+
+        const expected =
+            `{
+    "hello": "world"
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("comment out key/value pair", () => {
+        const text =
+            `{
+    /*"hello": "world"*/
+    "from": "json"
+}`;
+
+        const expected =
+            `{
+    "from": "json"
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("multi-line comment at end of text", () => {
+        const text =
+            `{
+    "hello": "world"
+}/**/`;
+
+        const expected =
+            `{
+    "hello": "world"
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("ignore multi-line comment inside string", () => {
+        const text =
+            `{
+    "hello": "wo/**/rld"
+}`;
+
+        const expected =
+            `{
+    "hello": "wo/**/rld"
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("ignore BOM", () => {
+        const text =
+            `\uFEFF{
+    "hello": "world"
+}`;
+
+        const expected =
+            `{
+    "hello": "world"
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("ignore trailing comma in object member list", () => {
+        const text =
+            `{
+    "obj": {
+        "hello": "world",
+        "from": "json",
+    }
+}`;
+
+        const expected =
+            `{
+    "obj": {
+        "hello": "world",
+        "from": "json"
+    }
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
+
+    test("ignore trailing comma in array element list", () => {
+        const text =
+            `{
+    "array": [
+        "element1",
+        "element2",
+    ]
+}`;
+
+        const expected =
+            `{
+    "array": [
+        "element1",
+        "element2"
+    ]
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
```

Comment:
Just thought of an issue. This won't work when a trailing comma has a comment at the end of the line. Fixing...